TopPlaces/TopPlaces/Model/PlacesPhotosProvider.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2016 Lightricks. All rights reserved.
+// Created by Gennadi Iosad.
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// Abstraction for a place description that can be presented to the user
+@protocol PlaceInfo<NSObject>
+
+
+/// Returns the URL to download the list of the photos from the place,
+/// the list length is bounded by \c maxLength
+- (NSURL*)urlOfPhotoInfoArrayWithMaxLength:(NSUInteger)maxLength;
+
+/// Compares to another instance of PlaceInfo conforming object
+- (NSComparisonResult)compare:(id<PlaceInfo>)other;
+
+/// Returns the title of the place
+@property (nonatomic, readonly) NSString *title;
+
+/// Returns additional details of the place
+@property (nonatomic, readonly) NSString *details;
+
+/// Returns the country of the place
+@property (nonatomic, readonly) NSString *country;
+
+@end
+
+
+
+/// Photo information absraction
+@protocol PhotoInfo<NSCoding>
+
+/// Returns the title of the image
+@property (nonatomic, readonly) NSString *title;
+
+
+/// Returns the details - auxillary info of the image
+@property (nonatomic, readonly) NSString *details;
+
+/// Returns the url for downloading the image
+@property (nonatomic, readonly) NSURL *url;
+
+@end
+
+
+/// Places and image lists provider abtraction
+@protocol PlacesPhotosProvider
+
+/// Download the places list
+- (NSURLSessionDownloadTask *)downloadPlacesWithCompletionHandler:
```

Comment:
Few comments regarding this protocol:
1) Regarding the methods name: we usually just name it XXXWithCompletion: for brevity (not CompletionHandler).
2) We tend to define the completion blocks prototypes with typedef, this makes the code a lot clearer.
3) I would avoid from returning an <code>NSURLSessionDownloadTask</code> here since this assumes a specific implementation of the methods. Instead I would return some more abstract interface (like NSOperation or even more simple then that).